### ISSUE TITLE
CLP: FE combine page-specific and reusable Q&As under the same header & accordion group

### ISSUE THREAD

**Issue by jilladams** [department-of-veterans-affairs/va.gov-cms#10940]
## Status
**[2024-05-06]** The BE ticket has been released #17805 
**[2024-04-24]** This ticket blocked by a BE ticket that's in the current sprint.

## Description
Q&As can be added to CLPs in 2 ways: 
1. Page-specific paragraph type
2. Reusable Q&A content type, as an entity reference. 

When we added Reusable Q&A, these were added in separate accordions: https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/layouts/campaign_landing_page.drupal.liquid#L463.  

Laura flagged that re: A11y, these would ideally present in the same accordion, with headers addressed. 

## Status

[2024-01-11] Solution identified during refinement proposed to Jordan, who signed off. Jill Slacked Danielle to get her sign off, at which time ticket will be updated and moved to refined.

## Eng notes
Related tickets: 
* #15717 - open ticket for known work we need to do around headers both inside and outside accordions for Q&As
* #10939 - where we added Reusable Q&As to CLPs
* #14814 - previous work around headers

## Design note
We did not create a Front-end design for adding Reusable Q&As to CLPs. We should make sure to consult @thejordanwood with plans moving forward.

## ACs
- [ ] Vet that displaying page-specific and reusable Q&As in the same accordion is technically feasible, given how each type of Q&A is built / referenced in the template
  - [ ] If so: review & update ACs as needed with PM, A11y, Design
  - [ ] If so: update points assessment after review, and cut a separate ticket to implement if not possible to complete the work within scope of this ticket
  - [ ] If not, flag for Product / A11y and agree on plan to document accordingly in the CLP

**Comment by jilladams** [comment]
@laflannery FYI for following along

**Comment by jilladams** [comment]
From refinement: 

Original treatment of Page-specific Q&As includes a hard coded H2: "Frequently asked questions"
Reusable Q&As come with an optional Header field, that will populate an H2. 

![Screenshot 2024-01-10 at 11 36 15 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/85581471/853cea25-e455-4055-99ad-768415ec1c22)

**Goal:** One header, with all Q&As listed below it (We probably need to verify this goal with CAIA?)
**Proposed solution:** 
* If a CLP contains a Reusable Q&A group with a Header: show that Header as the H2, with all Q&As below it (both Page Specific and Reusable). 
* If no Reusable Q&As, or if Reusable Q&As but the optional Reusable Q&A Header field is empty: show the hard coded Frequently Asked Questions H2

**Comment by jilladams** [comment]
Let's start with Jordan -- hey @thejordanwood does the proposal in the comment above seem reasonable from a design perspective?

**Comment by thejordanwood** [comment]
@jilladams This sounds like a good solution to me from a design perspective.

**Comment by jilladams** [comment]
Nice! Pushed this Q to Danielle T for feedback: https://dsva.slack.com/archives/C52CL1PKQ/p1704929589032639?thread_ts=1703712906.383219&cid=C52CL1PKQ

**Comment by DanielleThierryUSDSVA** [comment]
Hi! This makes sense to me for the solution. Thank you!

**Comment by randimays** [comment]
@DanielleThierryUSDSVA Is there a possibility for Campaign Landing Pages to have multiple reusable Q&A sections (with headers) added?

Currently, we don't have any CLPs with reusable Q&A sections added. But it is possible for the page to have multiple labeled reusable Q&A sections

<img width="852" alt="Screenshot 2024-04-04 at 6 22 51 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/19175324/fcae6db7-28f4-4848-9ec4-36757d6df697">

CC @thejordanwood @jilladams 

Do we want to determine how this scenario should be handled?

**Comment by randimays** [comment]
Adding to the above comment: if we do want to determine how to handle that scenario, we would also need to account for a mix of filled and blank headers in any particular order for multiple reusable Q&A groups (i.e. the first one is empty, the second one has a header, the third is empty, etc.)

**Comment by jilladams** [comment]
From scrum: 
* In prod, we have no Reusable Q&As on CLPs. We are thinking into the future.
* The proposed path doesn't account for multiple Reusable Q&A groups with or without headers. 
* If we opt to require the optional Reusable Q&A group header, then what do we do with Page-specific Q&As?

So maybe: we are thinking too far ahead, and should put this back in the backlog. Laura vote: She is auditing _all_ products we haven't touched for a11y purposes. There may be a case to make for a design review to follow those a11y reviews, to figure out which products need love, in general. 

For now: could limit to 1 Reusable Q&A group. Would require Drupal work. Reusable Q&A groups are reused across content types, so depends how it's used in Drupal. Looking into how RQA work on CLPs and we'll come back to this.

**Comment by jilladams** [comment]
Daniel discovered: we can limit the cardinality to 1 Reusable Q&A group on CLPs. That will only affect CLPs bc of the how the field is used here. This is easy but will require a config PR, so needs its own time relative to Daniel's priorities. @FranECross are you able to make that ticket?

**Comment by FranECross** [comment]
I'll get the ticket made today, @jilladams !

**Comment by jilladams** [comment]
@randimays FYI the CMS change for this is in PR. When we limit to a single object, the shape of the GraphQL data changes. @dsasser 's notes on this are here, for your reference / if you have questions: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17805#issuecomment-2080992129

**Comment by FranECross** [comment]
@randimays I moved this to the Ready column and removed the Blocked label. cc @jilladams 

I think this one is okay to move forward with, even considering that the design work is a priority, since it's following Daniel's recent ticket. Please let me know if either of you disagree @jilladams @randimays

**Comment by randimays** [comment]
@FranECross Sounds good to me 👍🏻 I opened a WIP PR with my older changes in case Josh is the one to pick this up.

**Comment by jilladams** [comment]
CMS Changes are in prod now.

**Comment by jilladams** [comment]
Unblocked: https://dsva.slack.com/archives/C076XFT3VCN/p1717775787969569

**Comment by randimays** [comment]
We tested this extensively in a Tugboat pre-merge and Laura reviewed and approved. We won't be able to test this in production since we don't have any reusable Q&As on CLPs right now. Closing as complete, but if anyone feels like we need to do an extra sanity check with staging data, please raise the flag here.
